app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -168,13 +168,14 @@ def associate_default_locations
         end
       end
 
-      def content_service(content_type)
+      def content_service(content_type, pulp3_support = false)
         if content_type.is_a?(String)
           content_type = RepositoryTypeManager.find_content_type(content_type)
         end
 
         #this will rely on smart proxy capabilities when available
-        content_type.pulp2_service_class
+        # Return pulp3 or 2 based on content type
+        pulp3_support ? content_type.pulp3_service_class : content_type.pulp2_service_class
```

Comment:
would it be possible for this method to just know which one to call?  if the content type knew its repository_type, it could look it up that way

Reply:
meaning, if you know the content_type, you could lookup the repository_type and figure out the pulp2 vs pulp3 from that

Reply:
Updated to look like this:
```
content_pulp3_support?(content_type)? content_type.pulp3_service_class : content_type.pulp2_service_class
```
``` content_pulp3_support?(content_type) ``` being the new method to see if the content type is pulp3 supported. The pulp3_support method takes repository as argument and we can possibly change that but this looks clearer I guess.. 
Was this what you meant?